src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/BitbucketDeploymentFactoryImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.atlassian.bitbucket.jenkins.internal.deployments;
+
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeploymentEnvironment;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.DeploymentState;
+import com.google.common.annotations.VisibleForTesting;
+import hudson.model.Job;
+import hudson.model.Result;
+import hudson.model.Run;
+import jenkins.branch.MultiBranchProject;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+
+import javax.annotation.CheckForNull;
+import java.util.Arrays;
+import java.util.Collection;
+
+public final class BitbucketDeploymentFactoryImpl implements BitbucketDeploymentFactory {
+
+    private static final Collection<Result> successfulResults = Arrays.asList(Result.SUCCESS, Result.UNSTABLE);
+
+    private final DisplayURLProvider displayURLProvider;
+
+    public BitbucketDeploymentFactoryImpl() {
+        this(DisplayURLProvider.get());
+    }
+
+    @VisibleForTesting
+    BitbucketDeploymentFactoryImpl(DisplayURLProvider displayURLProvider) {
+        this.displayURLProvider = displayURLProvider;
+    }
+
+    @Override
+    public BitbucketDeployment createDeployment(Run<?, ?> run, BitbucketDeploymentEnvironment environment) {
+        return createDeployment(run, environment, null);
+    }
+
+    @Override
+    public BitbucketDeployment createDeployment(Run<?, ?> run,
+                                                BitbucketDeploymentEnvironment environment,
+                                                @CheckForNull DeploymentState state) {
+        Job<?, ?> job = run.getParent();
+        String description = getDescription(run);
+        String key = job.getFullName();
+        String name = getName(job);
+        state = state == null ? getStateFromRun(run) : state;
+        String url = displayURLProvider.getRunURL(run);
+
+        return new BitbucketDeployment(run.getNumber(), description, name, environment, key, state, url);
+    }
+
+    private String getDescription(Run<?, ?> run) {
+        String description = run.getDescription();
```

Comment:
I hope this is a big field. Jenkins allows around 200k for this. and it can be formatted text (html or markdown, depending on configuration). Are we sure this is what we want? Does it make sense? (not objecting just pointing things out)